rust/src/common.rs

```diff
@@ -55,3 +55,20 @@ pub unsafe extern "C" fn rs_cstring_free(s: *mut c_char) {
     }
     drop(CString::from_raw(s));
 }
+
+/// Take ownership of `obj`, box it, and returns it's raw pointer value
```

Comment:
@jasonish we recently talked about our use of `Box` and whether it was necessary. Can't remember the conclusion. Any thoughts on that, and if we still need it on this commit?

Reply:
I believe its still needed in this case.  If we look at rs_dns_log_json_answer_v1() where a JsonBuilder is returned.  If we didn't box it and convert it into raw, I believe Rust would drop it at the end of that function.  Doing the process of Box::into_raw relinguishes ownership of it.

And I think we need to box it first so we get an actual valid pointer to the data, otherwise Rust is free to move it around to another location.

However, I'm not sure how keen I am on this wrapper function. In the DNS code when returning something to C, I like to see explicitly how its being done.

Reply:
Ok thanks for the Box explanation. I think I do like using util funcs like this however. I think it makes clear what happens w/o overloading ppl with the details.